### ISSUE TITLE
Are there any insurmountable obstacles hot-loading java dependencies from maven central?

### ISSUE THREAD

**Issue by vxe** [clojure-emacs/clj-refactor.el#413]
Essentially to be able to call 
``` elisp
cljr-add-project-dependency
```
or something like it and pull in dependencies from maven central as well as clojars.  The Hot loading dependency feature of this project is fantastic and the only reason I took a second look at Clojure.  Having to kill my repl session and restart Cider to pull in a Java dep is my final pain point.

This project is amazing, keep up the good work.

**Comment by expez** [comment]
Hi, and thanks for the kind words! ;)

> Are there any insurmountable obstacles hot-loading java dependencies from maven central?

No, there's no fundamental problems with this.  Everything is just a jar after all.

It's been a while since I wrote this feature, but the only reason maven central as a whole isn't exposed is because their API was different from clojars  and would likely require a different workflow.

clojars lets you grab everything and we can then present a completing read to the user, which is pretty sweet imo.  maven central didn't let you do this at the time--and in any event it might not be a good idea because there are a gazillion artifacts up there--so I limited the search of maven central to a few key clojure related organizations.

To add the proposed feature we probably have to ask for a query string, call out to maven central and then let the user perform a completing read on the result.

I'd very much welcome a pull-request if you want to dig into this :+1:
